doc/index.rst

```diff
@@ -67,13 +67,155 @@ Celeritas to transport on GPU.
 
 .. [#celeritas_vers] This documentation is generated from Celeritas |release|.
 
+Background
```

Comment:
Assuming a lot of these changes are pulled from/the same as in #1562 ?

Reply:
Yes, I moved all the documentation changes (aside from the `inp`-specific ones) over to make editing easier to manage.